server/src/main/java/org/elasticsearch/script/ScriptMetadata.java

```diff
@@ -189,84 +183,30 @@ public static ScriptMetadata fromXContent(XContentParser parser) throws IOExcept
-                            if (source.getLang().equals(Script.DEFAULT_TEMPLATE_LANG)) {
-                                deprecationLogger.critical(
-                                    DeprecationCategory.TEMPLATES,
-                                    "empty_templates",
-                                    "empty templates should no longer be used"
-                                );
-                            } else {
-                                deprecationLogger.critical(
-                                    DeprecationCategory.TEMPLATES,
-                                    "empty_scripts",
-                                    "empty scripts should no longer be used"
-                                );
-                            }
-                        }
-                    }
-
-                    exists = scripts.get(id);
-
-                    if (exists == null) {
-                        scripts.put(id, source);
-                    } else if (exists.getLang().equals(lang) == false) {
-                        throw new IllegalArgumentException("illegal stored script, id [" + id + "] used for multiple scripts with " +
-                            "different languages [" + exists.getLang() + "] and [" + lang + "]; scripts using the old namespace " +
-                            "of [lang#id] as a stored script id will have to be updated to use only the new namespace of [id]");
-                    }
-
-                    id = null;
-
                     break;
                 case START_OBJECT:
                     if (id == null) {
                         throw new ParsingException(parser.getTokenLocation(),
                             "unexpected token [" + token + "], expected [<id>, <code>, {]");
                     }
 
-                    exists = scripts.get(id);
-                    source = StoredScriptSource.fromXContent(parser, true);
-
-                    if (exists == null) {
-                        // due to a bug (https://github.com/elastic/elasticsearch/issues/47593)
-                        // scripts may have been retained during upgrade that include the old-style
-                        // id of lang#id; these scripts are unreachable after 7.0, so they are dropped
-                        if (id.contains("#") == false) {
-                            scripts.put(id, source);
+                    StoredScriptSource source = StoredScriptSource.fromXContent(parser, true);
+                    // as of 8.0 we drop scripts/templates with an empty source
+                    // this check should be removed for the next upgradable version after 8.0
+                    // since there is a guarantee no more empty scripts will exist
+                    if (source.getSource().isEmpty()) {
+                        if (Script.DEFAULT_TEMPLATE_LANG.equals(source.getLang())) {
+                            deprecationLogger.critical(DeprecationCategory.TEMPLATES, "empty_templates",
```

Comment:
I think these should be warning logs, not deprecations. The deprecation already happened, this is a warning that we are dropping something meaningless.

Reply:
Done.